src/script/components/LexicalInput/components/Mention/Mention.tsx

```diff
@@ -0,0 +1,212 @@
+
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import {useLexicalNodeSelection} from '@lexical/react/useLexicalNodeSelection';
+import {mergeRegister} from '@lexical/utils';
+import {
+  $getNodeByKey,
+  $getSelection,
+  $isDecoratorNode,
+  $isElementNode,
+  $isNodeSelection,
+  $isTextNode,
+  $setSelection,
+  BLUR_COMMAND,
+  CLICK_COMMAND,
+  COMMAND_PRIORITY_LOW,
+  GridSelection,
+  KEY_ARROW_LEFT_COMMAND,
+  KEY_ARROW_RIGHT_COMMAND,
+  KEY_BACKSPACE_COMMAND,
+  KEY_DELETE_COMMAND,
+  NodeKey,
+  NodeSelection,
+  RangeSelection,
+  SELECTION_CHANGE_COMMAND,
+} from 'lexical';
+
+// import {IS_IOS} from './environment';
+import {$isBeautifulMentionNode} from '../../nodes/MentionNode';
+import {getNextSibling, getPreviousSibling} from '../../utils/mention-utils';
+
+interface BeautifulMentionComponentProps {
+  mention: string;
+  nodeKey: NodeKey;
+  className?: string;
+  classNameFocused?: string;
+}
+
+export const Mention = (props: BeautifulMentionComponentProps) => {
+  const {mention, className = '', classNameFocused = '', nodeKey} = props;
+  const [editor] = useLexicalComposerContext();
+  const [isSelected, setSelected, clearSelection] = useLexicalNodeSelection(nodeKey);
+  const [selection, setSelection] = useState<RangeSelection | NodeSelection | GridSelection | null>(null);
+  const isFocused = $isNodeSelection(selection) && isSelected;
+  const ref = useRef(null);
+
+  const classNameFinal = useMemo(() => {
+    const classes = [className];
+    if (isFocused) {
+      classes.push(classNameFocused);
+    }
+    return classes.join(' ').trim() || undefined;
+  }, [className, classNameFocused, isFocused]);
+
+  const onDelete = useCallback(
+    (payload: KeyboardEvent) => {
+      if (isSelected && $isNodeSelection($getSelection())) {
+        payload.preventDefault();
+        const node = $getNodeByKey(nodeKey);
+        if ($isBeautifulMentionNode(node)) {
+          node.remove();
+        }
+        setSelected(false);
+      }
+      return false;
+    },
+    [isSelected, nodeKey, setSelected],
+  );
+
+  const onArrowLeftPress = useCallback(
+    (event: KeyboardEvent) => {
+      const node = $getNodeByKey(nodeKey);
+      if (!node || !node.isSelected()) {
+        return false;
+      }
+      let handled = false;
+      const nodeToSelect = getPreviousSibling(node);
+      if ($isElementNode(nodeToSelect)) {
+        nodeToSelect.selectEnd();
+        handled = true;
+      }
+      if ($isTextNode(nodeToSelect)) {
+        nodeToSelect.select();
+        handled = true;
+      }
+      if ($isDecoratorNode(nodeToSelect)) {
+        nodeToSelect.selectNext();
+        handled = true;
+      }
+      if (nodeToSelect === null) {
+        node.selectPrevious();
+        handled = true;
+      }
+      if (handled) {
+        event.preventDefault();
+      }
+      return handled;
+    },
+    [nodeKey],
+  );
+
+  const onArrowRightPress = useCallback(
```

Comment:
to avoid code duplication between left and right arrow key function we can pass an extra param called direction. and handle it based on it.
```

const handleArrowKeyEvent = useCallback(
  (event: KeyboardEvent, direction: 'left' | 'right') => {
  
  const node = $getNodeByKey(nodeKey);

    if (!node || !node.isSelected()) {
      return false;
    }

    let handled = false;
    let nodeToSelect: Node | null;

    if (direction === 'left') {
      nodeToSelect = getPreviousSibling(node);
    } else {
      nodeToSelect = getNextSibling(node);
    }

    if (nodeToSelect) {
      if ($isElementNode(nodeToSelect)) {
        if (direction === 'left') {
          nodeToSelect.selectEnd();
        } else {
          nodeToSelect.selectStart();
        }
        handled = true;
      } 
      // rest of the code
      }
}
```

Reply:
this has not been addressed.